### PR TITLE
I18n export import

### PR DESCRIPTION
**Pull request by vincentfretin**
see translations/README.md for the documentation

### CODE REVIEW

**Comment by vincentfretin** [comment]
The ./scripts/merge_po_files.bash needs the msgmerge command. On debian/ubuntu `apt-get install gettext` to have it. On mac, `brew install gettext` maybe? I don't know.

**Comment by vincentfretin** [comment]
For French, we can still use ./app/utils/translations.js for now, but we should move all French translations to a  po file when all our feature branches are merged to avoid conflicts.

File: assembl/static2/js/app/utils/deepen.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export default function deepen(o) {
```
Comment (cedricmessiant) [inline assembl/static2/js/app/utils/deepen.js]: a unit test would be welcome and would document this function ;-)

File: assembl/static2/scripts/translations2messages.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const fs = require('fs');
+const translations = require('../js/app/utils/translations');
+
+const englishTranslations = translations.en;
+
+function flatten(arr) {
```
Comment (cedricmessiant) [inline assembl/static2/scripts/translations2messages.js]: we have `lodash` in the dependencies, maybe we could use https://lodash.com/docs/4.17.4#flatten ?

File: assembl/static2/scripts/translations2messages.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  return arr.reduce((flat, toFlatten) => {
+    return flat.concat(Array.isArray(toFlatten) ? flatten(toFlatten) : toFlatten);
+  }, []);
+}
+
+const object2messages = (fullkey, objOrString) => {
```
Comment (cedricmessiant) [inline assembl/static2/scripts/translations2messages.js]: a unit test would be nice to document this one too

File: assembl/static2/js/app/utils/deepen.js (hunk 1 above)
Comment (ayyazdaniaryan) [inline assembl/static2/js/app/utils/deepen.js]: Would agree. A few jest tests and test of edge cases would be cool. 👍

File: assembl/static2/tests/unit/utils/deepen.spec.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import deepen from '../../../js/app/utils/deepen';
```
Comment (ayyazdaniaryan) [inline assembl/static2/tests/unit/utils/deepen.spec.js]: Oh sweet!

File: assembl/static2/translations/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    npm run i18n:export
+
+This generates an intermediate file static2/messages.json from which we generate the
+pot file using react-intl-po (but we don't use react-intl or
+babel-plugin-react-intl in this project).
+
+The `i18n:export` command takes care of resyncing the existing `locale/assembl-v2/*.po`
+files with the pot file with the msgmerge command.
+To have the msgmerge command on debian/Ubuntu `apt-get install gettext`, and on
+macOS `brew install gettext; brew link gettext`.
```
Comment (ayyazdaniaryan) [inline assembl/static2/translations/README.md]: Should likely add these to fabric so auto-installed instances of Assembl are ready to go for this. But good to know.
